Makefile.core.mk

```diff
@@ -82,11 +82,21 @@ endif
 # Typically same as GOPATH/bin, so tests work seemlessly with IDEs.
 
 export ISTIO_BIN=$(GOBIN)
-# Using same package structure as pkg/
 
 export ISTIO_OUT:=$(TARGET_OUT)
 export ISTIO_OUT_LINUX:=$(TARGET_OUT_LINUX)
 
+# Determine if one of the targets contains `test`. If it does, and we are
+# running in the container, make sure we point at the linux (container OS) binaries.
+ifeq ($(findstring test,${MAKECMDGOALS}),test)
```

Comment:
I don't think this is the right approach. Now my "make attest" target will fail :slightly_smiling_face: (not really a target, just an example). And for targets that are not called test but run binaries things will also probably break? Overall this seems like a workaround for the specific target but not a general fix that will apply to future changes and may cause some confusion.

if we do want to go with an approach like this, would it make more sense for these targets to explicitly declare some target, say `fix-container-out` and set these vars, rather than doing it like this?

I fully admit I haven't looked into this as much as you, so let me know if what I said doesn't make sense

Reply:
@howardjohn I don't understand your proposal - would you mind expanding?

How about we inverse this...

The only time we **NEED** arch-specific binaries is during `make build` when BUILD_WITH_CONTAINER=1. Otherwise, we always need Linux-specific binaries - any test cases, any docker cases etc.

I suggest we default all targets to TARGET_OUT_LINUX - except when people ae running build. If people are running `make build` on MacOS, both a TARGET_OUT_LINUX and TARGET_OUT are built...

Reply:
@sdake I did have a Slack conversation with @howardjohn about the issue. I agree that catching all targets with _test_ in them is probably not ideal, especially if we get multiple targets working. Also with the inverse solution, we might want to do something other than the _build_ target, and expect a host OS binary (I hate to say `make istioctl` but that seems like it might be pretty common).

The recap of my conversation with John was to go back to my earlier solution with updating tests that need to access an actual binary (*envoy* might be the only one) to actually use the env variable that points to the actual OS we are running in tree (at least that's how I read the conversation). I'm going to go back to by code that I was using to experiment with that. There is one change that is needed to fix `exec` messages. I'm now seeing an issue in the iptables tests that I'm investigating (output does not match with golden file). If that resulting changes are minimal, it does get us around messing with changes for specific targets, and pushing the change to only those tests that fail.

Reply:
@howardjohn @sdake  I did push an alternative PR #20485 which updates the test cases and their associated targets to use the LOCAL_OUT variable. If that one makes more sense, we can close this one.

Reply:
@ericvn @howardjohn rather than having private Slack conversations, can we keep review commentary on github so that other reviewers have the same context as those that would be involved in the private slack conversation?

Cheers
-steve